modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -596,14 +596,24 @@ public IgniteKernal(@Nullable GridSpringResourceContext rsrcCtx) {
 
     /** {@inheritDoc} */
     @Override public boolean isNodeInBaseline() {
-        ClusterNode locNode = localNode();
+        ctx.gateway().readLockAnyway();
```

Comment:
Why this change?

Reply:
`localNode()` takes the gw lock internally and the latter throws an exception if state is not `STARTED`. As a result the test `ReadMetricsOnNodeStartUpTest` was failing.
Rechecked it now and it doesn't reproduce anymore. Most likely due to the fact that I moved the call to `registerMetrics()` past the place where gw state is set to `STARTED`.

So I reverted the change for isNodeBaselne() back to original.